theano/sparse/basic.py

```diff
@@ -832,12 +832,24 @@ def perform(self, node, (x, ), (out, )):
         out[0] = x.astype(self.out_type)
 
     def grad(self, inputs, outputs_gradients):
-        if inputs[0].dtype in tensor.continuous_dtypes:
-            gz = outputs_gradients[0]
-            return [Cast(inputs[0].dtype)(gz)]
-        else:
-            return [None]
+        gz = outputs_gradients[0]
 
+        if gz.dtype in complex_dtypes:
+            raise NotImplementedError("grad not implemented for complex types")
+        if inputs[0].dtype in complex_dtypes:
+            raise NotImplementedError("grad not implemented for complex types")
```

Comment:
In the case of this op, can't the complex dtypes be handles the same way as the "float dtypes"